core/domain/state_domain.py

```diff
@@ -977,6 +1899,24 @@ def validate(
             raise utils.ValidationError(
                 'Hint(s) must be specified if solution is specified')
 
+        # These validations should only be run when we are validating for
+        # an exploration because as of now(16 Sept 2022) the data related
+        # to Question model is not yet fixed.
+        if validation_from_exploration:
+            interaction_id_to_validation_dict = {
+                'Continue': self._validate_continue_input,
```

Comment:
There is a separation between interaction and state, the state provides structure to add any interaction, and the interaction layer defines the structure of any specific interaction. I don't think this is maintainable i.e, state explicitly handling/validating different types of interaction. Why can't we have such validations in the Interaction layer i.e, either in interaction_spec.json (As a schema validation for customization arg specs) or inside extensions/interaction/InteractionName/Interaction.py file OR any other way to allow interactions to validate its cust_arg/rule_specs? is there any issue with that?

Reply:
@DubeySandeep is right, this can't be checked in like this. This is permanent code and if you write it like this here then it's going to just get added on to by other contributors and the codebase will not be maintainable.

Questions are relatively recent, so there shouldn't actually be validation errors. Also this is only run on-save, right? So existing questions will still be loadable.

Given that, I recommend that you put this code in the proper place (and ditto for any other code that is in the wrong place). Otherwise we are introducing serious maintainability issues to the codebase and that will cause problems in the long run.

Reply:
Thanks for your thoughts, @seanlip! 

@lkbhitesh07 Let me know if you need any help finding the correct place for these validations and can help you do it quickly!

Reply:
Apologies -- I think I retract part of my earlier comment.

The part about validating state interactions in general, regardless of whether we are doing it with questions or explorations, still stands. I also think that we should rename `interaction_id_to_validation_dict` to `interaction_id_to_validation_func`.

However, the part about putting it in the interaction files is trickier. Conceptually, yes, the code should go in each interaction and "core" code shouldn't refer to stuff in `extensions`. On the other hand, extensions/interactions is mainly meant to have declarative code -- notice how the interaction classes are all just static objects and there's no "logic" in the file. This starts breaking down, unfortunately, when you add validations etc., because the code becomes imperative and now the extensions folder needs to start knowing about domain object internal structures and that creates a tight coupling that we don't actually want.

The issue arises from it being difficult to encode some of these validations in a declarative way. I don't think it is feasible to do this in a short time period. Thus, I think that we should leave it as-is for now, but add a TODO/issue to find a way to encode these checks more declaratively.

@DubeySandeep if you see a declarative way to encode these that I am missing, please say, otherwise I would suggest we go with the above. What do you think?

Reply:
I have added the TODO.